std/file.d

```diff
@@ -570,6 +628,23 @@ void rename(RF, RT)(RF from, RT to)
     renameImpl(f, t, fromz, toz);
 }
 
+/// ditto
+void rename(RF, RT)(auto ref RF from, auto ref RT to)
+    if (isStringLike!RF || isStringLike!RT)
+{
+    import std.meta : staticMap;
+    alias selectString(R) = Select!(isStringLike!R, StringTypeOf!R, R);
+    alias Types = staticMap!(selectString, RF, RT);
+    rename!Types(from, to);
+}
```

Comment:
You get a similar result with this PR for `rename(1,1);`:
```
test.d(7): Error: template std.file.rename cannot deduce function from argument types !()(int, int), candidates are:
c:\s\d\rainers\phobos\std\file.d(610):        std.file.rename(RF, RT)(RF from, RT to) if ((isInputRange!RF && isSomeChar!(ElementEncodingType!RF) || isSomeString!RF) && !isStringLike!RF && (isInputRange!RT && isSomeChar!(ElementEncodingType!RT) || isSomeString!RT) && !isStringLike!RT)
c:\s\d\rainers\phobos\std\file.d(632):        std.file.rename(RF, RT)(auto ref RF from, auto ref RT to) if (isStringLike!RF || isStringLike!RT)
```

This shows the error is at the call site, but doesn't really offer an easy hint what's wrong as the trait does. I guess it's a lot more work to get the best of both, so I'm ok with this.